src/TriangleNodesProvisioner.cpp

```diff
@@ -73,6 +77,179 @@ namespace blitzdg {
+                gys = blitz::sum(dVMds(ii,jj)*yk(jj), jj);
+
+                gJac = gxr*gys - gxs*gyr;
+
+                // Invert the 2x2 mapping matrix.
+                grx = gys/gJac;
+                gry =-gxs/gJac;
+                gsx =-gyr/gJac;
+                gsy = gxr/gJac;
+
+                if (f == 0) { gnx = -gsx; gny = -gsy; }
+                else if (f == 1) { gnx = grx + gsx; gny = gry + gsy;}
+                else if (f == 2) { gnx = -grx; gny = -gry; }
+            
+                gsJ = blitz::sqrt(gnx*gnx + gny*gny);
+                gnx = gnx / gsJ;
+                gny = gny / gsJ;
+                gsJ = gsJ * gJac;
+
+                // copy to the global arrays.
+                for (index_type ig=0; ig < NGauss; ++ig) {
+                    nx(ids1(ig), k)  = gnx(ig);
+                    ny(ids1(ig), k)  = gny(ig);
+                    sJ(ids1(ig), k)  = gsJ(ig);
+                    Jac(ids1(ig), k) = gJac(ig);
+                    rx(ids1(ig), k)  = grx(ig);
+                    ry(ids1(ig), k)  = gry(ig);
+                    sx(ids1(ig), k)  = gsx(ig);
+                    sy(ids1(ig), k)  = gsy(ig);
+                }
+            
+                const index_vector_type& E2E = Mesh2D.get_EToE(), E2F = Mesh2D.get_EToF();
+                index_type k2 = E2E(NumFaces*k + f), f2 = E2F(NumFaces*k + f);
+
+                index_vector_type ids2(NGauss);
+                for (index_type ind=NGauss-1; ind >= 0; --ind) {
+                    ids2(ind) = f2*ind; 
+                }
+                if (k != k2) {
+                    for (index_type ig=0; ig < NGauss; ++ig) {
+                        mapP(ids1(ig) + k*NGauss) = mapM(ids2(ig) + k2*NGauss);
+                    }
+                } else {
+                    for (index_type ig=0; ig < NGauss; ++ig) {
+                        index_type bcFaceNode = mapM(ids1(ig) + k*NGauss);
+                        mapP(ids1(ig) + k*NGauss) = bcFaceNode;
+                        mapB.push_back(bcFaceNode);
+
+                        gbcMap[bcVec(NumFaces*k + f)].push_back(bcFaceNode);
+                    }
+                }
+            }
+        }
+
+        // get x & y grids on Gauss face mesh.
+        real_matrix_type gx(NumFaces*NGauss, NumElements), gy(NumFaces*NGauss, NumElements);
+
+        gx = blitz::sum(gaussInterpMatrix(ii, kk)*x(kk, jj), kk);
+        gy = blitz::sum(gaussInterpMatrix(ii, kk)*y(kk, jj), kk);
+
+        // Form vectorized quadrature weights.
+        real_matrix_type W(NumFaces*NGauss, NumElements);
+        real_matrix_type wmat(NGauss, NumElements);
+
+        W = 0.0;
+        wmat = w(ii) * (0*jj + 1.0);
+        W(Range(0, NGauss - 1),        Range::all()) = wmat;
+        W(Range(NGauss, 2*NGauss - 1), Range::all()) = wmat;
+        W(Range(2*NGauss, 3*NGauss-1), Range::all()) = wmat;
+
+        // Jacobian scaling.
+        W *= sJ;
+
+        return GaussFaceContext2D (
```

Comment:
It seems unnecessary to give each parameter its own line in the constructor call. Also, I would prefer the constructor call using `{}` brackets instead of `()`.